### PR TITLE
pc - add documentation and sample for .env

### PR DESCRIPTION
**Pull request by pconrad**
In this PR, we add documentation and a sample for the `.env` file.

We also add a few convenience items to the `.gitignore` for emacs/vim/MacOS

### CODE REVIEW

File: .env.SAMPLE [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Copy this file to .env (which is in the .gitignore)
+# See README.md for instructions on obtaining values
+
+AUTH0_DOMAIN=copy-value-from-auth0
+AUTH0_CLIENT_ID=copy-value-from-auth0
+AUTH0_CLIENT_SECRET=copy-value-from-auth0
+
+MONGODB_URI=mongodb+srv://adminuser:ZUONH9nRtRniyBRC@cluster0-6c3fw.mongodb.net/test?retryWrites=true&w=majority
```
Comment (scottpchow23) [inline .env.SAMPLE:8]: Do we want the live mongodb uri here?

Comment (pconrad) [reply .env.SAMPLE:8]: No, we do not.

Comment (pconrad) [reply .env.SAMPLE:8]: Though that is actually not a live address.

**Comment by pconrad** [comment]
Will address concerns above and reopen on master

**Pull request closed without merging**